File: home_assignments/java/TicketSeller/TicketSellerPersistence/src/main/java/ticketseller/persistence/interfaces/IRepository.java
package ticketseller.persistence.interfaces;

import ticketseller.model.entities.Identifiable;

import java.util.Collection;

public interface IRepository<ID, E extends Identifiable<ID>> {

    void add(E entity);

    void delete(E entity);

    void update(E entity, ID id);

    E findByID(ID id);

    Iterable<E> findAll();

    Collection<E> getAll();

}

File: home_assignments/csharp/TicketSeller/persistence/interfaces/IRepository.cs
using System.Collections.Generic;
using model.entities;

namespace persistence.interfaces
{
    public interface IRepository<TId, TEntity> where TEntity : Entity<TId>
    {
        void Add(TEntity festival);
        
        void Delete(TEntity festival);
        
        void Update(TEntity festival, TId id);

        TEntity FindById(TId id);

        IEnumerable<TEntity> FindAll();
        
        ICollection<TEntity> GetAll();
    }
}